instructions/include/instr.hpp

```diff
@@ -0,0 +1,63 @@
+
+/*
+ * Brief: bit field representation of an instructions
+ * w/ opcode const, and utility functions for managing instruction
+ *
+ * instruction bit fields are a fundamental component of the CPU
+ * emulator
+ *
+ */
+
+#ifndef INSTR_HPP
+#define INSTR_HPP
+
+// example of possible instruction layout
+#include <cstdint>
+
+#pragma "packed"
+struct instruction {
+  uint16_t opcode : 4;
+  uint16_t reg1 : 4;
```

Comment:
may want to call our variables mem1,2,3 etc now with mem-mem design

Reply:
updated